stdlib/http/src/test/java/org/ballerinalang/stdlib/utils/Services.java

```diff
@@ -85,13 +87,13 @@ public static HttpCarbonMessage invoke(int listenerPort, HTTPTestRequest request
             properties = Collections.singletonMap(HttpConstants.SRC_HANDLER, srcHandler);
         }
 
-        // Object[] signatureParams = HttpDispatcher.getSignatureParameters(resource, request,
-        // (MapValue) connectorEndpoint.get(SERVICE_ENDPOINT_CONFIG));
         Object[] signatureParams = HttpDispatcher.getSignatureParameters(resource, request, null);
         callback.setRequestStruct(signatureParams[0]);
 
         ObjectValue service = resource.getParentService().getBalService();
-        Executor.submit(service, resource.getName(), callback, properties, signatureParams);
+        Scheduler scheduler = httpServicesRegistry.getScheduler();
+        Executor.submit(scheduler, service, resource.getName(), callback, properties, signatureParams);
+        Executors.newSingleThreadExecutor().submit(scheduler::start);
         callback.sync();
```

Comment:
Do we still need this callback.sync()?

Reply:
Yes, That callback mocks the transport(a client) which does the executor submit in the test suite. Since we do not flow through HTTP transport in the unit tests, this call back waits there until ballerina respond to this callback to assert the response.